cardano-testnet/src/Testnet/Components/SPO.hs

```diff
@@ -401,3 +405,35 @@ registerSingleSpo identifier tap@(TmpAbsolutePath tempAbsPath') nodeConfigFile s
               poolColdVkeyFp
               currentRegistedPoolsJson
   return (poolId, poolColdSkeyFp, poolColdVkeyFp, vrfSkeyFp, vrfVkeyFp)
+
+-- | Generates Stake Pool Operator (SPO) voting files, using @cardano-cli@.
+--
+-- Returns a list of generated @File VoteFile In@ representing the paths to
+-- the generated voting files.
+generateVoteFiles :: (MonadTest m, MonadIO m, MonadCatch m)
+  => ConwayEraOnwards era -- ^ The conway era onwards witness for the era in which the
+                          -- transaction will be constructed.
+  -> H.ExecConfig -- ^ Specifies the CLI execution configuration.
+  -> FilePath -- ^ Base directory path where the voting files and directories will be
+              -- stored.
+  -> String -- ^ Name for the subfolder that will be created under 'work' to store
+            -- the output voting files.
+  -> String -- ^ Transaction ID string of the governance action.
+  -> Word32 -- ^ Index of the governance action.
+  -> [(PoolNodeKeys, [Char])] -- ^ List of tuples where each tuple contains a 'PoolNodeKeys'
+                              -- representing the SPO keys and a 'String' representing the
+                              -- vote type (i.e: "yes", "no", or "abstain").
+  -> m [File VoteFile In]
+generateVoteFiles ceo execConfig work prefix governanceActionTxId governanceActionIndex allVotes = do
+  baseDir <- H.createDirectoryIfMissing $ work </> prefix
+  forM (zip [(1 :: Integer)..] allVotes) $ \(idx, (spoKeys, vote)) -> do
```

Comment:
```suggestion
  H.forConcurrently_ (zip [(1 :: Integer)..] allVotes) $ \(idx, (spoKeys, vote)) -> do
```
that should give some speedup

Reply:
I've tried that when unifying with the `SPO` stuff, but couldn't because there was an issue with `forConcurrently_` requiring and providing `IO` and `execCli'` requiring `MonadTest` (I think it was `MonadTest`, if not something like that), and `IO` by itself is not `MonadTest`, so I am not sure about how to bridge that.

Reply:
I could add the `concurrency` package maybe? But anyway, for 3-10 votes it didn't make a noticeable difference in performance, in my experiments.

Reply:
[`forConcurrently_`](https://hackage.haskell.org/package/lifted-async-0.10.2.5/docs/Control-Concurrent-Async-Lifted.html#v:forConcurrently_) from `Hedgehog.Extras.Test.Concurrent` is an unlifted reexport from `lifted-async`, which can be used with `MonadTest`. I'll leave it up to you whether it's worth using it here.

Reply:
I see, I just had to add `MonadBase` to the type signatures.
Fixed here: https://github.com/IntersectMBO/cardano-node/pull/5809/commits/b604c4ad7341a8dd8cc8a76cb1cf3c62ff476732